include/aspect/particle/property/interface.h

```diff
@@ -36,6 +36,130 @@ namespace aspect
   {
     namespace Property
     {
+      /**
+       * This class is used to store all the necessary information to translate
+       * between the data structure of the particle properties (a flat vector of
+       * doubles) and the semantic meaning of these properties. It contains
+       * information about the three layers of particle property information:
+       *
+       * By 'property plugins' we mean each separate class that is derived from
+       * aspect::Particle::Property::Interface<dim>, and that is selected in
+       * the input file. This means in any model there are as many property
+       * plugins as entries in the 'List of tracer properties' input parameter.
+       *
+       * Each plugin can create one or more 'property fields'. Property fields
+       * are interpreted as distinctly named particle properties. Most plugins
+       * contain only one field, but some group distinctly named properties
+       * into groups, such as the 'InitialComposition' plugin, which creates
+       * one property field per compositional field and names the property
+       * fields according to the compositional field names. When writing particle
+       * data to output files each 'property field' will be written into a
+       * separate output field.
+       *
+       * Last each field can contain several 'property components'
+       * if it represents a vector or tensor property. These components can
+       * not be named individually, but it is still important to be able to
+       * know how many components belong to a particular field.
+       *
+       * Information that is often required by other algorithms is for example
+       * the number of components (= number of doubles in the property vector)
+       * of a particle field or plugin, and its position within the particle
+       * property vector. All of this information might be required within
+       * loops over all fields, or for a specific field either identified by
+       * its index, or by its name.
+       *
+       * @ingroup ParticleProperties
+       */
+      class ParticlePropertyInformation
+      {
+        public:
+          /**
+           * Empty default constructor.
+           */
+          ParticlePropertyInformation();
+
+          /**
+           * Constructor. Initialize the various arrays of this structure with the
+           * given property information collected from the individual plugins.
+           */
+          ParticlePropertyInformation(const std::vector<std::vector<std::pair<std::string,unsigned int> > > &property_information);
```

Comment:
I suspect that I know what the argument's nested levels all mean, but it would probably still be useful to explain.